feat: BROS-594: Add Distribution row to Task Summary

hlomzik opened the pull request:
## Aggregation/Distribution

It displays aggregations for choices, taxonomies, etc.:
- For number and rating it shows average number
- For labels and taxonomies it shows counts for every option
- For choices it shows distribution of every choice, including annotations where this choice was skipped

## Styles

A lot of styles were fixed in Task Summary: badges, table design, zebra rows, sticky column, usernames, etc.

## Autoselect of Summary tab

When you switch between tasks you'll end up in the same pane you've been before — either annotation view, Summary tab or Compare tab. It makes it easier to review multiple tasks.

We already had a memory about a tab in View All, but now we also remember was it annotation view or View All. Technically it's tricky because we exit View All in a lot of places where we select annotation. And we select all annotations (or one with FF_SIMPLE_INIT on) on LSF init, so it always resets the view. So now we only exit View All if we are already in View All and do an interaction with UI.

<img width="1088" height="618" alt="Screenshot 2025-11-13 at 18 02 44" src="https://github.com/user-attachments/assets/010dc541-217f-4f38-ba3d-befab00991d8" />


<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

hlomzik committed:
feat: BROS-594: Add Distribution row to Task Summary

It displays aggregations for choices, taxonomies, etc.:
- For number and rating it shows average number
- For choices, labels and taxonomies it shows counts for every option

Also a lot of styles were fixed in Task Summary.

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19347995250)
> Successfully merged:  create mode 100644 label_studio/fsm/urls.py

hlomzik committed:
Merge branch 'develop' into 'fb-bros-594/summary-distribution'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19347995250

hlomzik committed:
Remove link to non-existent ChatRegion to fix `yarn docs`

hlomzik committed:
Merge branch 'fb-bros-594/summary-distribution' of github.com:HumanSignal/label-studio into fb-bros-594/summary-distribution

hlomzik commented:
/fmt
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19348703150)
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19348703150)

hlomzik committed:
Remember View All selection going from task to task

hlomzik committed:
Fix the issue in lsf-sdk

If there are no annotations don't fail trying to save the draft

bmartel commented on web/libs/editor/src/components/TaskSummary/Chip.tsx:
Do we not have a similar Badge component in the UI lib which could be used here?

hlomzik replied:
This Chip has its own display logic. We might use `Badge` as display component inside `Chip`, but styles are different, we work on this later with @ricardoantoniocm

bmartel commented on web/libs/editor/src/components/TaskSummary/utils.ts:
Similarly it seems like we have many ways we handle color cycling for labels, it might be worthwhile consolidating the logic to a shared core util so everywhere we ever need this label cycling, it is the same.

hlomzik replied:
Good idea in general, but we have different logic for labels. Plus it uses outdated libraby we don't control, so it would be a nice effort to make all this controllable and consolidated, but definitely not in this PR.

hlomzik committed:
Improve styles of the aggregation row

hlomzik committed:
Fix shadows

hlomzik committed:
Show percentage again instead of counts

hlomzik committed:
Small improvements for Aggregation/Distribution row

- Move all code to one dedicated file
- Don't show horizontal scroll on small tables
- Don't show expand button for small aggregations
- Simplify code a little

hlomzik committed:
Fix linting

hlomzik committed:
Fix linting one more time

hlomzik committed:
Replace deprecated param for biome

We should only apply it to schema/tags.json

hlomzik committed:
Fix tests

Also make `showEmpty` true by default

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19432922680)
> Successfully merged:  create mode 100644 web/libs/datamanager/src/components/Filters/types/TaskStateFilter.jsx

hlomzik committed:
Merge branch 'develop' into 'fb-bros-594/summary-distribution'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19432922680
